build/android/src/net/minetest/minetest/MinetestAssetCopy.java

```diff
@@ -95,70 +97,75 @@ protected String doInBackground(String... files)
 			m_filenames    = new Vector<String>();
 			m_tocopy       = new Vector<String>();
 			m_asset_size_unknown = new Vector<String>();
-			String baseDir = 
-					Environment.getExternalStorageDirectory().getAbsolutePath()
-					+ "/";
-			
-			
+
+			Context con = getApplicationContext();
+			String baseDir = con.getFilesDir().getAbsolutePath() + "/";
+//			String baseDir =
+//					Environment.getExternalStorageDirectory().getAbsolutePath()
+//					+ "/";
+
+
 			// prepare temp folder
-			File TempFolder = new File(baseDir + "Minetest/tmp/");
-			
+			File TempFolder = new File(baseDir
+				+ MtNativeActivity.ProjectName + "/tmp/");
```

Comment:
I only want to use proper paths, not make platform code use platform independent code for no reason.
I'll use the cache dir though.